airflow/api_fastapi/common/parameters.py

```diff
@@ -281,6 +282,30 @@ def _safe_parse_datetime(date_to_check: str) -> datetime:
         )
 
 
+class _WarningTypeFilter(BaseParam[str]):
+    """Filter on warning type."""
+
+    def to_orm(self, select: Select) -> Select:
+        if self.value is None and self.skip_none:
+            return select
+        return select.where(DagWarning.warning_type.contains(self.value))
+
+    def depends(self, warning_type: DagWarningType | None = None) -> _WarningTypeFilter:
+        return self.set_value(warning_type)
+
+
+class _DagIdInDagWarningFilter(BaseParam[str]):
```

Comment:
I wrote this as the other one `_DagIdPatternSearch` will look for any pattern. If we are ok to reuse it, then we can update it to also take which Model to pass.

Reply:
IT is operating on the `dag_id` and `DagWarning` <=> `Dag` are linked by the foreign key.

I think sqlalchemy is smart enough to make an implicit joing if you do the filter on `DagModel.dag_id`. Otherwise you can just join on `Dag` so you can simply re-use the `_DagIdPatternSearch`.

Also we can update the name `xxxxINxxxxx` makes reader think that is is an `array_contain` filter but it's just a normal equal.

Reply:
Updated it to be a `_DagIDFilter`